Tribler/Core/Modules/MetadataStore/OrmBindings/collection_node.py

```diff
@@ -0,0 +1,416 @@
+            ).delete()
+            orm.flush()  # Just in case...
+            if not children or 0 not in children:
+                return {}
+            return children
+
+        @staticmethod
+        @db_session
+        def get_commit_forest():
+            children = db.CollectionNode.get_children_dict_to_commit()
+            if not children:
+                return {}
+            # We want a separate commit tree/queue for each toplevel channel
+            forest = {}
+            toplevel_nodes = [node for node in children.pop(0)]
+            for root_node in toplevel_nodes:
+                # Tree -> stack -> queue
+                commit_queue = []
+                tree_stack = [root_node]
+                while tree_stack and children.get(tree_stack[-1].id_, None):
+                    # Traverse the tree from top to bottom converting it to a stack
+                    while children.get(tree_stack[-1].id_, None):
+                        node = children[tree_stack[-1].id_].pop()
+                        tree_stack.append(node)
+
+                    while not issubclass(type(tree_stack[-1]), db.CollectionNode):
+                        commit_queue.append(tree_stack.pop())
+                    # Unwind the tree stack until either the stack is empty or we meet a non-empty node
+                    while tree_stack and not children.get(tree_stack[-1].id_, None):
+                        while not issubclass(type(tree_stack[-1]), db.CollectionNode):
+                            commit_queue.append(tree_stack.pop())
+
+                        # It was a terminal collection
+                        collection = tree_stack.pop()
+                        commit_queue.append(collection)
+
+                if not commit_queue or commit_queue[-1] != root_node:
+                    commit_queue.append(root_node)
+                forest[root_node.id_] = tuple(commit_queue)
+
+            return forest
+
+        @staticmethod
+        def prepare_commit_queue_for_channel(commit_queue):
+            """
+            This routine prepares the raw commit queue for commit by updating the elements' properties and
+            re-signing them. Also, it removes the channel entry itself from the queue [:-1], because its
+            meaningless to put it in the blobs, as it must be updated with the new infohash after commit.
+
+            :param commit_queue:
+            :return:
+            """
+            for node in commit_queue:
+                # Avoid updating entries that must be deleted:
+                # soft delete payloads require signatures of unmodified entries
+                if issubclass(type(node), db.CollectionNode) and node.status != TODELETE:
+                    # Update recursive count of actual non-collection contents
+                    node.num_entries = select(
+                        (g.num_entries if g.metadata_type == COLLECTION_NODE else 1) for g in node.actual_contents
```

Comment:
`1` just means `1` here. It is not a constant, it just means that that there is only one sub-object to count. Of course, this could be changed to something like:
```python
DEFAULT_NUMBER_OF_OBJECTS_IN_NON_COLLECTION_NODE = 1
```
But I don't think it will make it more readable. I will just add a comment on this instead.

Reply:
Right